### PR TITLE
Huggingface pretrain + finetune notebook

### PR DESCRIPTION
**Pull request by dakinggg**
# What does this PR do?
Adds a notebook with a tutorial for pretraining and finetuning a hugging face model

# What issue(s) does this change relate to?
Closes [CO-1444](https://mosaicml.atlassian.net/browse/CO-1444?atlOrigin=eyJpIjoiYjJlNTQwZWZjNzYyNDRlZWIxYTE5Y2JiYzdjYjNhMTkiLCJwIjoiaiJ9)

# Before submitting
- [x] Have you read the [contributor guidelines](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md)?
- [x] Was this change discussed/approved in a GitHub issue first? It is much more likely to be merged if so.
- [x] Did you update any related docs and document your change?
- [x] Did you update any related tests and add any new tests related to your change? (see [testing](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#running-tests))
- [x] Did you run the tests locally to make sure they pass?
- [x] Did you run `pre-commit` on your change? (see the `pre-commit` section of [prerequisites](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#prerequisites))

### CODE REVIEW

File: examples/pretrain_finetune_huggingface.ipynb [hunk 1]
```diff
@@ -0,0 +1,496 @@
+{
```
Comment (mvpatel2000) [inline examples/pretrain_finetune_huggingface.ipynb:1]: <p>link is broken in NBViewer... is it on renderer or genuine bug?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mosaicml/composer/pull/1775/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

**Review by mvpatel2000 — APPROVED** [review]
LGTM. Approving since u need one from eng, but we should get a stamp from sam or alex as well

File: examples/pretrain_finetune_huggingface.ipynb (hunk 1 above)
Comment (dakinggg) [reply examples/pretrain_finetune_huggingface.ipynb:1]: The link works for me... https://app.reviewnb.com/mosaicml/composer/pull/1775?

Comment (mvpatel2000) [reply examples/pretrain_finetune_huggingface.ipynb:1]: image
this is what I see -- should there be links here?

Comment (dakinggg) [reply examples/pretrain_finetune_huggingface.ipynb:1]: oh thats what you meant. yeah those appear to be broken in nbviewer, they are ok for me locally

File: examples/huggingface_models.ipynb [hunk 2]
```diff
@@ -351,15 +351,18 @@
     "\n",
     "If you want to keep learning more, try looking through some of the documents linked throughout this tutorial to see if you can form a deeper intuition for what's going on in these examples.\n",
     "\n",
-    "In addition, please continue to explore our tutorials! Here are a couple suggestions:\n",
+    "In addition, please continue to explore our tutorials and examples! Here are a couple suggestions:\n",
     "\n",
-    "* Use Composer to pre-train a BERT model and fine-tune it on the GLUE benchmark with our [custom entrypoint][glue].\n",
+    "* Explore domain specific pretraining of a Hugging Face model in addition to finetuning it in a follow on [tutorial][tutorial]\n",
```
[the remark this answers is not in the document]
Comment (dakinggg) [reply examples/huggingface_models.ipynb:356]: oh yeah that sentence is super confusing, I will rewrite

**Review by samhavens — APPROVED** [review]
Other than that sentence it LGTM

**Pull request merged**